packages/neuron-wallet/src/services/addresses.ts

```diff
@@ -149,128 +160,196 @@ export default class AddressService {
       }
       return AddressService.toAddress(addressMetaInfo)
     })
-    const testnetChange = change.map(arr => arr[0])
-    const mainnetChange = change.map(arr => arr[1])
     return {
-      testnetReceiving,
-      mainnetReceiving,
-      testnetChange,
-      mainnetChange,
+      receiving,
+      change,
     }
   }
 
-  private static toAddress = (addressMetaInfo: AddressMetaInfo): AddressInterface[] => {
+  private static toAddress = (addressMetaInfo: AddressMetaInfo): AddressInterface => {
     const path: string = Address.pathFor(addressMetaInfo.addressType, addressMetaInfo.addressIndex)
-    const testnetAddress: string = addressMetaInfo.accountExtendedPublicKey.address(
-      addressMetaInfo.addressType,
-      addressMetaInfo.addressIndex,
-      AddressPrefix.Testnet
-    ).address
-
-    const mainnetAddress: string = addressMetaInfo.accountExtendedPublicKey.address(
+    const address: string = addressMetaInfo.accountExtendedPublicKey.address(
       addressMetaInfo.addressType,
       addressMetaInfo.addressIndex,
-      AddressPrefix.Mainnet
+      AddressService.getAddressPrefix()
     ).address
 
-    const addressToParse = NetworksService.getInstance().isMainnet() ? mainnetAddress : testnetAddress
-    const blake160: string = AddressParser.toBlake160(addressToParse)
+    const blake160: string = AddressParser.toBlake160(address)
 
-    const testnetAddressInfo: AddressInterface = {
+    const addressInfo: AddressInterface = {
       walletId: addressMetaInfo.walletId,
-      address: testnetAddress,
+      address,
       path,
       addressType: addressMetaInfo.addressType,
       addressIndex: addressMetaInfo.addressIndex,
-      txCount: 0,
-      liveBalance: '0',
-      sentBalance: '0',
-      pendingBalance: '0',
-      balance: '0',
-      blake160,
-      version: AddressVersion.Testnet,
+      blake160
     }
 
-    const mainnetAddressInfo = {
-      ...testnetAddressInfo,
-      address: mainnetAddress,
-      version: AddressVersion.Mainnet,
-    }
-
-    return [testnetAddressInfo, mainnetAddressInfo]
+    return addressInfo
   }
 
-  public static nextUnusedAddress = (walletId: string): AddressInterface | undefined => {
-    const addressEntity = AddressDao.nextUnusedAddress(walletId,  AddressService.getAddressVersion())
-    if (!addressEntity) {
+  private static async maxAddressIndex(walletId: string, addressType: AddressType): Promise<number | undefined> {
+    const result = await getConnection()
+      .getRepository(HdPublicKeyInfo)
+      .createQueryBuilder()
+      .select('addressIndex')
+      .where({walletId, addressType})
+      .orderBy('addressIndex', 'DESC')
+      .getRawOne()
```

Comment:
`count` is enough

Reply:
I want it to ensure the max index is indeed the max one stored in the table, just in case there are corruptions in the table data.